packages/atomic/create-atomic/src/fetch-page.ts

```diff
@@ -0,0 +1,39 @@
+import fetch from 'node-fetch';
+import {
+  PageManifest,
+  PageManifestBody,
+  ResultsPlaceholder,
+} from './page-manifest.js';
+
+export async function fetchPageManifest(
+  platformUrl: string,
+  orgId: string,
+  pageId: string,
+  apiKey: string
+) {
+  const url = `${platformUrl}/rest/organizations/${orgId}/searchinterfaces/${pageId}/manifest/v1`;
+  const body: PageManifestBody = {
+    pagePlaceholders: {
+      results: ResultsPlaceholder,
+    },
+  };
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${apiKey}`,
+    },
+    body: JSON.stringify(body),
+  });
+  return replaceResultsPlaceholder((await response.json()) as PageManifest);
+}
+
+function replaceResultsPlaceholder(pageManifest: PageManifest): PageManifest {
+  const resultManagerComponent = '<results-manager></results-manager>';
+  pageManifest.markup = pageManifest.markup.replace(
+    pageManifest.results.placeholder!,
```

Comment:
😕 why is there a non-null assertion operator?
If there's always a placeholder, then the interface should reflect that.

Reply:
Hmm then, if I'm not mistaken you cannot be sure that `placeholder` is actually defined when this statement is reached, correct?